src/datasource/components/TimeRangeWarning.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { css } from '@emotion/css';
+import { GrafanaTheme2, TimeRange } from '@grafana/data';
+import { Icon, useStyles2 } from '@grafana/ui';
+import { TIME_RANGE_WARNING_THRESHOLD_DAYS } from '../constants';
+
+interface TimeRangeWarningProps {
+  timeRange?: TimeRange;
+}
+
+function getTimeRangeDurationDays(timeRange: TimeRange): number {
+  const from = timeRange.from.valueOf();
+  const to = timeRange.to.valueOf();
+  const durationMs = to - from;
+  return durationMs / (1000 * 60 * 60 * 24);
+}
+
+function isTimeRangeLarge(timeRange: TimeRange, thresholdDays: number = TIME_RANGE_WARNING_THRESHOLD_DAYS): boolean {
+  return getTimeRangeDurationDays(timeRange) >= thresholdDays;
+}
+
+function formatDuration(days: number): string {
+  if (days >= 365) {
```

Comment:
how did we know what to use for the limit of what is considered a large time range (1 year vs 90 days for example)? could we maybe add a comment for this?